utils/add_platform_rule.py

```diff
@@ -0,0 +1,321 @@
+#!/usr/bin/python3
+
+import argparse
+import subprocess
+import pathlib
+import sys
+import os
+import time
+
+'''
+ This script is intended to help content writers create a new application check for Kubernetes.
+ The 'create' subcommand creates the initial files for a new rule and fetches the raw URL of the object in question (unless you specify the URL).
+ The 'cluster-test' subcommand pushes the content to your cluster, and then runs a Platform scan for your rule with compliance-operator.
+ The 'test' subcommand builds your content locally and tests directly using an openscap podman container.
+
+ $ utils/add_platform_rule.py create --rule=ocp_proxy_has_ca --type="proxies.config" --name="cluster" --yamlpath=".spec.trustedCA.name" --match="[a-zA-Z0-9]*"
+ creating check for "/apis/config.openshift.io/v1/proxies/cluster" with yamlpath ".spec.trustedCA.name" satisfying match of "[a-zA-Z0-9]*"
+ wrote applications/openshift/ocp_proxy_has_ca/rule.yml
+ wrote applications/openshift/ocp_proxy_has_ca/oval/shared.xml
+
+ $ utils/add_platform_rule.py cluster-test --rule=ocp_proxy_has_ca
+ testing rule ocp_proxy_has_ca in-cluster
+ deploying compliance-operator
+ pushing image build to cluster
+ waiting for cleanup from previous test run
+ output from last phase check: LAUNCHING NOT-AVAILABLE
+ output from last phase check: RUNNING NOT-AVAILABLE
+ output from last phase check: AGGREGATING NOT-AVAILABLE
+ output from last phase check: DONE COMPLIANT
+ COMPLIANT
+'''
+
+PLATFORM_RULE_DIR = 'applications/openshift'
+OSCAP_TEST_IMAGE = 'quay.io/compliance-operator/openscap-ocp:1.3.3'
+OSCAP_CMD_OPTS = 'oscap xccdf eval --fetch-remote-resources --profile xccdf_org.ssgproject.content_profile_test --results-arf /tmp/report-arf.xml /content/ssg-ocp4-ds.xml'
+
+RULE_TEMPLATE = ('''prodtype: ocp4
+
+title: {TITLE}
+
+description: TBD
+
+rationale: TBD
+
+identifiers:
+    cce@ocp4: 84209-6
```

Comment:
What does this identifier mean? Is it OK for all generated rules to have the same?

Reply:
Oh right, It's the unique CCE id for the rule. So each rule needs one from shared/references/cce-redhat-avail.txt . When you add a rule you have to cut from that file. I added a flag to be able to fill in the reference, but leaving the value and removal from cce-redhat-avail.txt up to the user for now..